libcextract/SymbolExternalizer.cpp

```diff
@@ -513,6 +513,33 @@ bool SymbolExternalizer::FunctionUpdater::Update_References_To_Symbol(Stmt *stmt
       expr->setDecl(NewSymbolDecl);
       replaced = true;
     }
+  } else if (DeclStmt::classof(stmt)) {
+    /* To handle symbol renames on situations like the sizeof below
+     *
+     * static char x[4];
+     *
+     * void f(void) {
+     *    char y[sizeof(x)];
+     * }
+     */
+    DeclStmt *dstmt = (DeclStmt *) stmt;
+    Decl *ddecl = dstmt->getSingleDecl();
+
+    auto vec_of_ranges = Get_Range_Of_Identifier_In_SrcRange(ddecl->getSourceRange(),
+                                                             OldSymbolName.c_str());
+
+    if (vec_of_ranges.size() > 0) {
```

Comment:
We may iterate on all elements of `vec_of_ranges` because we can have things like `char y[sizeof(x)][sizeof(x)]`.